regolith/builders/publistbuilder.py

```diff
@@ -76,6 +76,26 @@ def latex(self):
                 cat_grants = cat_grants + "_" + g
             filestub = filestub + "{}".format(cat_grants)
             qualifiers = qualifiers + " from Grant{} {}".format(pl, text_grants)
+        if self.rc.kwargs:
+            kw = True
+            kwargs = self.rc.kwargs
+            if isinstance(kwargs, str):
+                key, value = kwargs.split(':', 1)  # Split only at the first colon
+                if key == 'facilities':
+                    facilities = value
+                    filestub = filestub + "_facility_{}".format(facilities)
+                    qualifiers = qualifiers + " from facility {}".format(facilities)
+                elif key == 'doi':
+                    doi = value
+                    filestub = filestub + "_doi_{}".format(doi)
+                    qualifiers = qualifiers + " from facility {}".format(doi)
+                else:
+                    facilities, doi = None, None
```

Comment:
we can probably turn 5 lines of code into one line of code if we set this at the top.  Fewer lines means less to maintain.

Reply:
Yes Professor. I can fix that now as well as the string. Not sure if you would like me to delete the elif key=="doi" part as well.